src/hotspot/share/oops/oop.inline.hpp

```diff
@@ -185,7 +185,7 @@ int oopDesc::size_given_klass(Klass* klass)  {
       // disjunct below to fail if the two comparands are computed across such
       // a concurrent change.
       assert((s == klass->oop_size(this)) ||
```

Comment:
It might be nicer to throw the second half of this assert into the .cpp file, like
assert(s == klass->oop_size(this) || gc_has_forwarded(), "wrong array object size");
With the special comment above it in the cpp file.  Then someone has the potential to add more information if the assert fails, and that gets Universe::hpp out of the inline file (if that helps with future cleanups).

Reply:
I'd like to keep this PR minimal. If I change `(Universe::heap()->is_gc_active() && is_objArray() && is_forwarded() && (get_UseParallelGC() || get_UseG1GC())` to `gc_has_forwarded()`, I would also need to move the comments about `UseParallelGC and UseG1GC can change the length field of an "old copy"...`. I think that should be done in a separate PR.